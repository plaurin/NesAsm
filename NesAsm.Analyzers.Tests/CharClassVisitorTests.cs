using ApprovalTests;
using ApprovalTests.Reporters;
using NesAsm.Analyzers.Visitors;

namespace NesAsm.Analyzers.Tests;

[UseReporter(typeof(VisualStudioReporter))]
public class CharClassVisitorTests
{
    [Fact]
    public void Test()
    {
        var result = CharClassVisitor.Process(@"C:\Users\pasca\Dev\GitHub\NesAsm\NesAsm.Example\game1.png");

        Approvals.Verify(result);
    }

    [Fact]
    public void TestImportCharScript()
    {
        TestScript("ImportCharDefinition.cs");
    }

    private static void TestScript(params string[] filenames)
    {
        var generator = new CharGenerator();

        var sourcePaths = filenames.Select(filename => $@"..\..\..\TestFiles\{filename}").ToArray();

        GeneratorTestUtilities.TestGenerator(generator, sourcePaths);
    }
}