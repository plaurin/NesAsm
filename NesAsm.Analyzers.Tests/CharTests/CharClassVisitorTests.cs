using ApprovalTests.Reporters;

namespace NesAsm.Analyzers.Tests.CharTests;

[UseReporter(typeof(VisualStudioReporter))]
public class CharClassVisitorTests
{
    [Fact]
    public void TestImportCharScript()
    {
        TestScript("ImportCharDefinition.cs");
    }

    [Fact]
    public void TestInvalidSizeCharScript()
    {
        TestScript("InvalidSize.cs");
    }

    [Fact]
    public void TestMoreThanFourPalettesCharScript()
    {
        TestScript("MoreThanFourPalettes.cs");
    }

    [Fact]
    public void TestMoreThanThreeNonDefaultColorCharScript()
    {
        TestScript("MoreThanThreeNonDefaultColor.cs");
    }

    [Fact]
    public void TestColorMismatchCharScript()
    {
        TestScript("ColorMismatch.cs");
    }

    private static void TestScript(params string[] filenames)
    {
        var generator = new CharGenerator();

        var sourcePaths = filenames.Select(filename => $@"..\..\..\CharTests\TestFiles\{filename}").ToArray();

        GeneratorTestUtilities.TestGenerator(generator, sourcePaths);
    }
}