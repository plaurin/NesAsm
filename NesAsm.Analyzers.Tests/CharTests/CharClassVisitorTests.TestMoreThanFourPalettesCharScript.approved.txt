- Main diagnostics:
..\..\..\CharTests\TestFiles\MoreThanFourPalettes.cs(6,2): info NA1002: Png image contains 10 different color palettes
- SyntaxTrees Count: 2
- Output diagnostics:
- RunResult.GeneratedTrees Count: 1
- RunResults diagnostics:
..\..\..\CharTests\TestFiles\MoreThanFourPalettes.cs(6,2): info NA1002: Png image contains 10 different color palettes
- RunResult[0].Generator: Microsoft.CodeAnalysis.IncrementalGeneratorWrapper
- RunResult[0] diagnostics:
..\..\..\CharTests\TestFiles\MoreThanFourPalettes.cs(6,2): info NA1002: Png image contains 10 different color palettes
- RunResult[0].GeneratedSources Count: 1
- RunResult[0].Exception: 
- RunResult[0].GeneratedSources[0].HintName: Asm.MoreThanFourPalettes.cs
- RunResult[0].GeneratedSources[0].SourceText:
/* Generator Asm code in file Asm.MoreThanFourPalettes.s
; Auto generated code using the NesAsm project
; Tile 0
.byte 0
.byte 0
.byte 0
.byte 0
.byte 0
.byte 0
.byte 0
.byte 0


.byte 0
.byte 0
.byte 0
.byte 0
.byte 0
.byte 0
.byte 0
.byte 0


; Tile 1
.byte 8
.byte 12
.byte 14
.byte 255
.byte 255
.byte 14
.byte 12
.byte 8


.byte 0
.byte 0
.byte 0
.byte 0
.byte 0
.byte 0
.byte 0
.byte 0


; Tile 2
.byte 16
.byte 48
.byte 112
.byte 255
.byte 255
.byte 112
.byte 48
.byte 16


.byte 0
.byte 0
.byte 0
.byte 0
.byte 0
.byte 0
.byte 0
.byte 0


; Tile 3
.byte 24
.byte 24
.byte 24
.byte 24
.byte 255
.byte 126
.byte 60
.byte 24


.byte 0
.byte 0
.byte 0
.byte 0
.byte 0
.byte 0
.byte 0
.byte 0


; Tile 4
.byte 24
.byte 60
.byte 126
.byte 255
.byte 24
.byte 24
.byte 24
.byte 24


.byte 0
.byte 0
.byte 0
.byte 0
.byte 0
.byte 0
.byte 0
.byte 0


; Tile 5
.byte 0
.byte 224
.byte 142
.byte 196
.byte 100
.byte 36
.byte 228
.byte 0


.byte 0
.byte 0
.byte 0
.byte 0
.byte 0
.byte 0
.byte 0
.byte 0


; Tile 6
.byte 0
.byte 224
.byte 142
.byte 200
.byte 110
.byte 40
.byte 238
.byte 0


.byte 0
.byte 0
.byte 0
.byte 0
.byte 0
.byte 0
.byte 0
.byte 0


; Tile 7
.byte 7
.byte 3
.byte 61
.byte 3
.byte 3
.byte 61
.byte 3
.byte 7


.byte 0
.byte 0
.byte 0
.byte 0
.byte 0
.byte 0
.byte 0
.byte 0


; Tile 8
.byte 60
.byte 60
.byte 231
.byte 195
.byte 255
.byte 255
.byte 195
.byte 195


.byte 0
.byte 0
.byte 0
.byte 0
.byte 0
.byte 0
.byte 0
.byte 0


; Tile 9
.byte 60
.byte 126
.byte 219
.byte 219
.byte 255
.byte 195
.byte 126
.byte 60


.byte 0
.byte 0
.byte 0
.byte 0
.byte 0
.byte 0
.byte 0
.byte 0


; Tile 10
.byte 0
.byte 102
.byte 126
.byte 255
.byte 255
.byte 126
.byte 60
.byte 24


.byte 0
.byte 0
.byte 0
.byte 0
.byte 0
.byte 0
.byte 0
.byte 0


; Palette 0
.byte 15
.byte 13
.byte 13
.byte 13

.byte 15
.byte 13
.byte 13
.byte 13

.byte 15
.byte 13
.byte 13
.byte 13

.byte 15
.byte 13
.byte 13
.byte 13

; Palette 1
.byte 15
.byte 1
.byte 13
.byte 13

.byte 15
.byte 1
.byte 13
.byte 13

.byte 15
.byte 1
.byte 13
.byte 13

.byte 15
.byte 1
.byte 13
.byte 13

; Palette 2
.byte 15
.byte 18
.byte 13
.byte 13

.byte 15
.byte 18
.byte 13
.byte 13

.byte 15
.byte 18
.byte 13
.byte 13

.byte 15
.byte 18
.byte 13
.byte 13

; Palette 3
.byte 15
.byte 3
.byte 13
.byte 13

.byte 15
.byte 3
.byte 13
.byte 13

.byte 15
.byte 3
.byte 13
.byte 13

.byte 15
.byte 3
.byte 13
.byte 13

; Palette 4
.byte 15
.byte 35
.byte 13
.byte 13

.byte 15
.byte 35
.byte 13
.byte 13

.byte 15
.byte 35
.byte 13
.byte 13

.byte 15
.byte 35
.byte 13
.byte 13

; Palette 5
.byte 15
.byte 32
.byte 13
.byte 13

.byte 15
.byte 32
.byte 13
.byte 13

.byte 15
.byte 32
.byte 13
.byte 13

.byte 15
.byte 32
.byte 13
.byte 13

; Palette 6
.byte 25
.byte 15
.byte 13
.byte 13

.byte 25
.byte 15
.byte 13
.byte 13

.byte 25
.byte 15
.byte 13
.byte 13

.byte 25
.byte 15
.byte 13
.byte 13

; Palette 7
.byte 15
.byte 57
.byte 13
.byte 13

.byte 15
.byte 57
.byte 13
.byte 13

.byte 15
.byte 57
.byte 13
.byte 13

.byte 15
.byte 57
.byte 13
.byte 13

; Palette 8
.byte 15
.byte 17
.byte 13
.byte 13

.byte 15
.byte 17
.byte 13
.byte 13

.byte 15
.byte 17
.byte 13
.byte 13

.byte 15
.byte 17
.byte 13
.byte 13

; Palette 9
.byte 15
.byte 21
.byte 13
.byte 13

.byte 15
.byte 21
.byte 13
.byte 13

.byte 15
.byte 21
.byte 13
.byte 13

.byte 15
.byte 21
.byte 13
.byte 13


*/
